### ISSUE TITLE
plugin crashes homebridge when subdevice is unavailable

### ISSUE THREAD

**Issue by movy** [prasad-edlabadka/homebridge-tuya-ir#28]
whenever a subdevice is unavailable or deleted and plugin cannot access it, it crashes whole HB:

```js
[10/26/2021, 8:11:01 PM] [TuyaIR] Failed to get remote configuration for: eb2a4d5379838c1a61hxii
[10/26/2021, 8:11:01 PM] [TuyaIR] Server returned error: 'permission deny'
[10/26/2021, 8:11:01 PM] [TuyaIR] [{}]
[10/26/2021, 8:11:01 PM] TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received undefined
    at new NodeError (node:internal/errors:363:5)
    at Hash.update (node:internal/crypto/hash:105:11)
    at Object.generate (/usr/local/lib/node_modules/homebridge/node_modules/hap-nodejs/src/lib/util/uuid.ts:11:11)
    at /usr/local/lib/node_modules/homebridge-tuya-ir/src/platform.ts:89:42
    at /usr/local/lib/node_modules/homebridge-tuya-ir/src/lib/TuyaIRDiscovery.ts:25:17
    at /usr/local/lib/node_modules/homebridge-tuya-ir/src/lib/TuyaAPIHelper.ts:117:21
    at Request._callback (/usr/local/lib/node_modules/homebridge-tuya-ir/src/lib/TuyaAPIHelper.ts:297:13)
    at Request.self.callback (/usr/local/lib/node_modules/homebridge-tuya-ir/node_modules/request/request.js:185:22)
    at Request.emit (node:events:394:28)
    at Request.<anonymous> (/usr/local/lib/node_modules/homebridge-tuya-ir/node_modules/request/request.js:1154:10)
```
    
    I think some error catching would be welcome.

**Comment by prasad-edlabadka** [comment]
Will have a look. Though I have put error handling everywhere...looks like some corners are missed. :(

**Comment by prasad-edlabadka** [comment]
Fixed in release 0.1.18
